compiler/rustc_serialize/src/opaque.rs

```diff
@@ -718,3 +718,51 @@ impl<'a> serialize::Decodable<Decoder<'a>> for Vec<u8> {
         Ok(v)
     }
 }
+
+// An integer that will always encode to 8 bytes.
+pub struct IntEncodedWithFixedSize(pub u64);
+
+impl IntEncodedWithFixedSize {
+    pub const ENCODED_SIZE: usize = 8;
+}
+
+impl serialize::Encodable<Encoder> for IntEncodedWithFixedSize {
+    fn encode(&self, e: &mut Encoder) -> EncodeResult {
```

Comment:
Now that we have `Encoder::emit_raw_bytes()` and `u64::to_le_bytes()` these methods can be greatly simplified.